services/src/main/java/io/scalecube/services/Microservices.java

```diff
@@ -0,0 +1,238 @@
+ * <li>Provides fluent, java 8 lamda apis.</li>
+ * <li>embeddable and lightweight.</li>
+ * <li>utilizes completable futures but primitives and messages can be used as well completable futures gives the
+ * advantage of composing and chaining service calls and service results. or implementing SEDA architecture. tested
+ * basic performance roundtrip (request/response) latency</li>
+ * <li>low latency</li>
+ * <li>supports routing extensible strategies when selecting service endpoints</li>
+ * 
+ * </p><b>basic usage example:</b>
+ * 
+ * <pre>
+ * 
+ * <b><font color="green">//Define a serivce interface and implement it.</font></b>
+ * {@code
+ *    <b>{@literal @}Service</b>
+ *    <b><font color="9b0d9b">public interface</font></b> GreetingService {  
+ *
+ *         <b>{@literal @}ServiceMethod</b>
+ *         CompletableFuture<String> asyncGreeting(String string);
+ *     }
+ *    
+ *     <b><font color="9b0d9b">public class</font></b> GreetingServiceImpl implements GreetingService {
+ *
+ *       {@literal @}Override
+ *       <b><font color="9b0d9b">public</font></b> CompletableFuture<String> asyncGreeting(String name) {
+ *         <b><font color="9b0d9b">return</font></b> CompletableFuture.completedFuture(" hello to: " + name);
+ *       }
+ *     }
+ *     <b><font color="green">//Build a microservices cluster instance.</font></b>
+ *     Microservices microservices = Microservices.builder()
+ *       <b><font color="green">//Introduce GreetingServiceImpl pojo as a micro-service.</font></b>
+ *         .services(<b><font color="9b0d9b">new</font></b> GreetingServiceImpl())
+ *         .build();
+ * 
+ *     <b><font color="green">//Create microservice proxy to GreetingService.class interface.</font></b>
+ *     GreetingService service = microservices.proxy()
+ *         .api(GreetingService.class)
+ *         .create();
+ * 
+ *     <b><font color="green">//Invoke the greeting service async.</font></b>
+ *     CompletableFuture<String> future = service.asyncGreeting("joe");
+ * 
+ *     <b><font color="green">//handle completable success or error.</font></b>
+ *     future.whenComplete((result, ex) -> {
+ *      if (ex == <b><font color="9b0d9b">null</font></b>) {
+ *        // print the greeting.
+ *         System.<b><font color="9b0d9b">out</font></b>.println(result);
+ *       } else {
+ *         // print the greeting.
+ *         System.<b><font color="9b0d9b">out</font></b>.println(ex);
+ *       }
+ *     });
+ * }
+ * </pre>
+ */
+
+public class Microservices {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Microservices.class);
+  private static final ServiceProcessor serviceProcessor = new AnnotationServiceProcessor();
+
+  private final ICluster cluster;
+
+  private final ServiceRegistry serviceRegistry;
+
+  private final ServiceProxyFactory proxyFactory;
+
+  private final ServiceDispatcher localDispatcher;
+
+  private Microservices(ICluster cluster, Optional<Object[]> services, boolean isSeed) {
```

Comment:
I don't understand why you need optional here. Usually optional is not used with sequences because you can just use empty sequence. This will simplify code here, in ServiceRegistry